app/controllers/posts_controller.rb

```diff
@@ -0,0 +1,83 @@
+# frozen_string_literal: true
+
+class PostsController < ApplicationController
+  before_action :set_post, only: %i[show edit update destroy]
+  before_action :authenticate_user!
+  before_action :check_user, only: %i[edit update destroy]
+
+  # GET /posts
+  # GET /posts.json
+  def index
+    @posts = Post.all
```

Comment:
- [ ] `index` page shouldn't display all posts, but posts for self and friends only.